syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service ClientHandler {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  // Sends another greeting
  rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
  // Lists users
  rpc ListUsers (ListRequest) returns (ListReply) {}
  // Requests to Login
  rpc Login (LoginRequest) returns (LoginReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
  string message = 2;
}

message LoginRequest {
  string name = 1;
}

message ListRequest {
    string args = 1;
}

message SendMessageRequest {
  string name = 1;
  string message = 2;
  string target = 3;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message LoginReply {
  string message = 1;
  int32 status = 2;
}

message ListReply {
  int32 status = 1;
  string wildcard = 2;
  string message = 3;
  repeated string user = 4;
  
}
